services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/client_api.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import traceback
+from typing import Dict, Optional
+
+import httpx
+from fastapi import FastAPI
+
+from ...core.settings import DynamicSidecarSettings
+from ...models.schemas.dynamic_services import MonitorData
+from .errors import MonitorException
+
+logger = logging.getLogger(__name__)
+
+
+def get_url(dynamic_sidecar_endpoint: str, postfix: str) -> str:
+    """formats and returns an url for the request"""
+    url = f"{dynamic_sidecar_endpoint}{postfix}"
+    return url
+
+
+def log_httpx_http_error(url: str, method: str, formatted_traceback: str) -> None:
+    # mainly used to debug issues with the API
+    logging.debug(
+        (
+            "%s -> %s generated:\n %s\nThe above logs can safely "
+            "be ignored, except when debugging an issue "
+            "regarding the dynamic-sidecar"
+        ),
+        method,
+        url,
+        formatted_traceback,
+    )
+
+
+class DynamicSidecarClient:
+    """Will handle connections to the service sidecar"""
+
+    def __init__(self, app: FastAPI):
+        self._app: FastAPI = app
+        self._heatlth_request_timeout: httpx.Timeout = httpx.Timeout(1.0, connect=1.0)
+
+        dynamic_sidecar_settings: DynamicSidecarSettings = (
+            app.state.settings.dynamic_services.DYNAMIC_SIDECAR
+        )
+
+        self.httpx_client = httpx.AsyncClient(
+            timeout=httpx.Timeout(
+                dynamic_sidecar_settings.DYNAMIC_SIDECAR_API_REQUEST_TIMEOUT,
+                connect=1.0,
```

Comment:
Because you connect to a dynamic-sidecar (a service in the swarm), you generally 1 second is more then enough. This way it fails faster.
Will move to dynamic_sidecar_settings and add some explanation.